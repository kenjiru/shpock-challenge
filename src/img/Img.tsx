import * as React from "react";
import { Component, ReactElement } from "react";

class Img extends Component<IImgProps> {
    public render(): ReactElement<HTMLElement> {
        return (
            <img
                className={this.props.className}
                src={`img/${this.props.src}`}
            />
        );
    }
}

interface IImgProps {
    className?: string;
    src: string;
}

export default Img;
