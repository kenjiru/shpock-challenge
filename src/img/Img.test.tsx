import * as React from "react";
import { shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";

import Img from "./Img";

describe("ErrorMessage", () => {
    it("should render correctly", () => {
        const output = shallow(
            <Img className="foo" src="bar.png"/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
