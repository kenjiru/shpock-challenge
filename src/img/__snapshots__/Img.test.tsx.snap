// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly 1`] = `
<img
  className="foo"
  src="img/bar.png"
/>
`;
